pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -189,7 +189,8 @@ func (r *Reconciler) handleUpdate(ctx context.Context, placementObj fleetv1beta1
 			klog.ErrorS(err, "Failed to extract the resource index from the resourceSnapshot", "placement", placementKObj, "resourceSnapshot", latestResourceSnapshotKObj)
 			return ctrl.Result{}, controller.NewUnexpectedBehaviorError(err)
 		}
-		selectedResourceIDs, err = controller.CollectResourceIdentifiersUsingMasterResourceSnapshot(ctx, r.Client, placementObj.GetName(), latestResourceSnapshot, strconv.Itoa(latestResourceSnapshotIndex))
+		placementKey := controller.GetObjectKeyFromNamespaceName(placementObj.GetNamespace(), placementObj.GetName())
```

Comment:
CollectResourceIdentifiersUsingMasterResourceSnapshot probably should take the namespacedName instead of string, next PR.